darts/explainability/tft_explainer.py

```diff
@@ -0,0 +1,249 @@
+
+        Parameters
+        ----------
+        plot
+            Whether to plot the variable selection weight.
+
+        Returns
+        -------
+        TimeSeries
+            The variable selection weight.
+
+        """
+
+        if plot:
+            # plot the encoder and decoder weights
+            self._plot_cov_selection(
+                self.encoder_importance,
+                title="Encoder variable importance",
+            )
+            self._plot_cov_selection(
+                self.decoder_importance,
+                title="Decoder variable importance",
+            )
+
+        return {
+            "encoder_importance": self.encoder_importance,
+            "decoder_importance": self.decoder_importance,
+        }
+
+    def explain(self, **kwargs) -> ExplainabilityResult:
+        """Returns the explainability result of the TFT model.
+
+        The explainability result contains the attention heads of the TFT model.
+        The attention heads determine the contribution of time-varying inputs.
+
+        Parameters
+        ----------
+        kwargs
+            Arguments passed to the `predict` method of the TFT model.
+
+        Returns
+        -------
+        ExplainabilityResult
+            The explainability result containing the attention heads.
+
+        """
+        super().explain()
+        # without the predict call, the weights will still bet set to the last iteration of the forward() method
+        # of the _TFTModule class
+        if "n" not in kwargs:
+            kwargs["n"] = self._model.model.output_chunk_length
+
+        _ = self._model.predict(**kwargs)
+
+        # get the weights and the attention head from the trained model for the prediction
+        attention_heads = (
+            self._model.model._attn_out_weights.squeeze().sum(axis=1).detach()
+        )
+
+        # return the explainer result to be used in other methods
+        return ExplainabilityResult(
+            {
+                0: {
+                    "attention_heads": TimeSeries.from_dataframe(
+                        pd.DataFrame(attention_heads).transpose()
+                    ),
+                }
+            },
+        )
+
+    @staticmethod
+    def plot_attention_heads(
+        expl_result: ExplainabilityResult,
+        plot_type: Optional[Literal["all", "time", "heatmap"]] = "time",
+    ):
+        """Plots the attention heads of the TFT model."""
+        attention_heads = expl_result.get_explanation(
+            component="attention_heads",
+            horizon=0,
+        )
+        if plot_type == "all":
+            fig = plt.figure()
+            attention_heads.plot(
+                label="Attention Head",
+                plot_all_components=True,
+                figure=fig,
+            )
+            # move legend to the right side of the figure
+            plt.legend(bbox_to_anchor=(0.95, 1), loc="upper left")
+            plt.xlabel("Time steps in past")
```

Comment:
I don't find it immediately clear that the higher these time steps the further ago the corresponding lags are.
How about maybe relabeling as `"Time steps in the past (# lags)"` ? This also applies to other places where this is used as an x-label.

Reply:
I have changed the axis label to `"Time steps in the past (# lags)"`